traffic_portal/app/src/common/modules/form/user/form.user.tpl.html

```diff
@@ -25,6 +25,7 @@
             <li class="active">{{userName}}</li>
         </ol>
         <div class="pull-right" role="group" ng-show="!settings.isNew">
+            <a class="btn btn-primary" ng-href="/#!/change-logs?username={{userName}}">View Change Log</a>
```

Comment:
unfortunately, for whatever reason the ngHref directive won't sanitize for you, so the value here should be inserted via a function that produces a sanitized output from the username using `encodeURIComponent`. Alternatively, you could create a simple AngularJS filter for query string parameters named something like `encodeURIComponent`, for example, then apply it here. That'd look like this in a file in `traffic_portal/app/src/common/filters/`:
```javascript
/**
 * This is a factory function for an AngularJS filter meant to allow easy
 * sanitization of interpolated input into URL query strings.
 * 
 * @example
 * 
 * <a ng-href="https://example.test/something?foo={{someString | encodeURIComponent}}">
 *   A safe link
 * </a>
 * 
 * @returns {(input: string) => string} A filter that sanitizes input text for safe insertion into URL query strings.
 */
 function EncodeURIComponentFilter() {
	return function(input) {
		return encodeURIComponent(input);
	};
};

module.exports = EncodeURIComponentFilter;
```
Then, just add a line to `traffic_portal/app/src/common/filters/index.js` like the other filters have, something like
```javascript
// in the @example above this is the name I used, but it can be anything
.filter("encodeURIComponent", require("./EncodeURIComponentFilter.js")),
```
(assuming the filter was written to `EncodeURIComponentFilter.js` - but it can be any file name).
Finally, just like the example, pass the data through the pipe in the `ng-href`:
```html
<a class="btn btn-primary" ng-href="/#!/change-logs?username={{userName | encodeURIComponent}}">View Change Log</a>
```

This is roughly one more line of work than just encoding it in a function in the users form controller, but it would allow us to re-use it everywhere it's needed without any extra work. Doesn't matter which you choose, though.

Reply:
Copy!